### PR TITLE
feat(environments): add new hooks and data-accesses

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1044

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> You are going to thank you later if it contains tests

File: libs/domains/environments/data-access/src/lib/domains-environments-data-access.ts [hunk 1]
```diff
@@ -30,7 +41,7 @@ export const environments = createQueryKeys('environments', {
       return new Promise<{ state: RunningState } | null>(() => {})
     },
   }),
-  detail: (environmentId: string) => ({
+  details: (environmentId: string) => ({
```
Comment (RemiBonnet) [inline libs/domains/environments/data-access/src/lib/domains-environments-data-access.ts:44]:
```suggestion
  details: ({ environmentId }: { environmentId: string }) => ({
```

File: libs/domains/environments/feature/src/lib/hooks/use-environment/use-environment.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export interface UseEnvironmentProps {
 export function useEnvironment({ environmentId }: UseEnvironmentProps) {
   return useQuery({
     // eslint-disable-next-line @typescript-eslint/no-extra-non-null-assertion
-    ...queries.environments.detail(environmentId!!),
+    ...queries.environments.details(environmentId!!),
```
Comment (RemiBonnet) [inline libs/domains/environments/feature/src/lib/hooks/use-environment/use-environment.ts:11]:
```suggestion
    ...queries.environments.details({ environmentId: environmentId!! }),
```

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
I added two feedback on comment + I don't know if it's relative to the migration but during my tests I see this problem: https://www.loom.com/share/129667bbcecf484c99957f3a408e5535

File: libs/domains/environments/data-access/src/lib/domains-environments-data-access.ts (hunk 1 above)
Comment (ctjhoa) [reply libs/domains/environments/data-access/src/lib/domains-environments-data-access.ts]: https://github.com/Qovery/console/pull/1233/commits/863b9d76cb88f8e4cc961e8bac9ed2eacd081200

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**